crates/db/src/lib.rs

```diff
@@ -308,6 +307,82 @@ impl DB {
         Ok(res)
     }
 
+    pub async fn get_erc721_tokens(
+        &self,
+        chain_id: u32,
+        owner: Address,
+    ) -> Result<Vec<Erc721Token>> {
+        let res: Vec<Erc721Token> = sqlx::query(
+            r#" SELECT *
+        FROM erc721_tokens
+        WHERE chain_id = ? AND owner = ?"#,
+        )
+        .bind(chain_id)
+        .bind(format!("0x{:x}", owner))
+        .map(|row| row.try_into().unwrap())
+        .fetch_all(self.pool())
+        .await?;
+
+        Ok(res)
+    }
+
+    pub async fn get_erc721_collections(
+        &self,
+        chain_id: u32,
+        address: Address,
+    ) -> Result<Vec<Erc721Collection>> {
+        let res: Vec<Erc721Collection> = sqlx::query(
+            r#" SELECT *
+        FROM erc721_collections
+        WHERE contract IN (SELECT DISTINCT contract FROM erc721_tokens WHERE owner = ? AND chain_id =?)"#,
+        )
+        .bind(format!("0x{:x}", address))
+        .bind(chain_id)
+        .map(|row| row.try_into().unwrap())
+        .fetch_all(self.pool())
+        .await?;
+
+        Ok(res)
+    }
+
+    pub async fn get_erc721_missing_data(&self, chain_id: u32) -> Result<Vec<Erc721Token>> {
+        let res: Vec<_> = sqlx::query(
+            r#"SELECT *
+        FROM erc721_tokens
+        WHERE chain_id = ? AND uri IS NULL"#,
+        )
+        .bind(chain_id)
+        .map(|row| row.try_into().unwrap())
+        .fetch_all(self.pool())
+        .await?;
+
+        Ok(res)
+    }
+
+    pub async fn save_erc721_data(
+        &self,
+        address: Address,
+        chain_id: u32,
+        token_id: U256,
+        owner: Address,
+        name: String,
+        symbol: String,
+        uri: String,
+        metadata: String,
```

Comment:
Linter is showing an error here
`error: this function has too many arguments (9/7)`